File: src/main/java/com/example/chatService/entity/FriendRequest.java
package com.example.chatService.entity;

import com.example.chatService.dto.FriendStatus;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter @Setter
public class FriendRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "from_user_id")
    private User fromUser;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "to_user_id")
    private User toUser;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private FriendStatus status;


}

File: src/main/java/com/example/chatService/entity/FriendRequest.kt
package com.example.chatService.entity

import com.example.chatService.dto.FriendStatus
import jakarta.persistence.*

@Entity
class FriendRequest(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "from_user_id")
    var fromUser: User? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "to_user_id")
    var toUser: User? = null,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var status: FriendStatus? = null
)
